walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -12,12 +12,103 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Figure out what everything is 
+// Find events for attendies, to show conflicts
+// Sort by start time
+// Search for conflicts
+// TODO: Find times where attendies can meet without conflict (Time ranges for mettings without conflict)
+
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  // @param Event (Collection) - All the events existing on an organization calendar: name, time range, attendies...
+
+  // @param Meeting request - Meeting of a specific time with certain attendies (and duration)
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+      // Collection of required attendees
+      // Collection of optional attendees
+
+      // COllection of required attendee times
+      // Collection of optional attendee times (findAttendeeEventTimes())
+      // Possibly all attendees -- times
+
+      // Collection of available times (required/optional)
+
+      // if/else available for all or available for required 
+
+      //for loop?
+      //Multiple methods... 
+
+      // Find who need to be at meetings...
+      // Find what meetings that are scheduled
+      // Find times
+      // If space in between meetings is >= duration of meeting... 
+      // add to a list of those time frames
+
+      // Returns a collection of time ranges
+
+      ArrayList<TimeRange> eventsForRequiredAttendees = findAttendeeEventTimes(events, request.getAttendees());
+      ArrayList<TimeRange> eventsForOptionalAttendees = findAttendeeEventTimes(events, request.getOptionalAttendees());
+
+      // ArrayList of all assigned and optional events
+      ArrayList<TimeRange> assignedAndOptionalEvents = new ArrayList();
+      assignedAndOptionalEvents.addAll(eventsForRequiredAttendees);
+      assignedAndOptionalEvents.addAll(eventsForOptionalAttendees);
+
+      // Sorting the actual events by start time
+      Collections.sort(eventsForRequiredAttendees, TimeRange.ORDER_BY_START);
+      Collections.sort(eventsForOptionalAttendees, TimeRange.ORDER_BY_START);
+      Collections.sort(assignedAndOptionalEvents, TimeRange.ORDER_BY_START);
+
+      // Return possible meeting times for required attendees
+      ArrayList<TimeRange> possibleMeetingTimes = 
+        findAppropriateTimes(assignedAndOptionalEvents, request.getDuration());
+
+      return possibleMeetingTimes;
+  }
+
+  // Lets get the times for events that attendees are going to.. 
```

Comment:
Method documentation should indicate in third person what the method does. "Gets the times for existing events that attendees are going to."

Also, avoid adding "..." to the end of your comments